src/Compilers/CSharp/Test/Semantic/Semantics/NullCoalescingAssignmentTests.cs

```diff
@@ -1343,5 +1334,75 @@ public static void Main()
                 Diagnostic(ErrorCode.ERR_ObjectRequired, "P").WithArguments("C.P").WithLocation(7, 9)
             );
         }
+
+        [Fact]
+        public void ThrowExpressionRHS()
+        {
+            CompileAndVerify(@"
+using System;
+public class C
+{
+    public static void Main()
+    {
+        try
+        {
+            object o = null;
+            o ??= throw new Exception();
+        }
+        catch (Exception)
+        {
+            Console.WriteLine(""Succeeded"");
+        }
+    }
+}
+", expectedOutput: "Succeeded");
+
+            CreateCompilation(@"
+public class C
+{
+    static object Property { get; }
+    public static void Main()
+    {
+        Property ??= throw null;
+    }
+}
+").VerifyDiagnostics(
+                // (7,9): error CS0200: Property or indexer 'C.Property' cannot be assigned to -- it is read only
+                //         Property ??= throw null;
+                Diagnostic(ErrorCode.ERR_AssgReadonlyProp, "Property").WithArguments("C.Property").WithLocation(7, 9)
+            );
+        }
+
+        [Fact]
+        public void TypeParameterLHS()
+        {
+            CreateCompilation(@"
+class C
+{
+    void M<T>(T t)
+    {
+        t ??= default;
+    }
+}
+").VerifyDiagnostics(
+                // (6,9): error CS0019: Operator '??=' cannot be applied to operands of type 'T' and 'default'
+                //         t ??= default;
```

Comment:
Seems like this should work.

Reply:
`??=` is a test for null. `T` is not constrained to be allowed to be `null` here, it's exactly the same semantics as `??`.

---
In reply to: [200767537](https://github.com/dotnet/roslyn/pull/28253#discussion_r200767537) [](ancestors = 200767537)

Reply:
`?.` works. If this is currently spec'd to do the same as `??` that's fine, but I don't see any reason it shouldn't work. We can talk about it in LDM.

Reply:
I've added it to my list of topics.